### PR TITLE
Andrew/menu

### PR DESCRIPTION
**Pull request by amillman**

### CODE REVIEW

File: sage-ios/SAGE/SAGE/AdminOperations.swift [hunk 1]
```diff
@@ -128,12 +135,19 @@ class AdminOperations {
...
             NetworkingConstants.kSortAttr: CheckinConstants.kTimeCreated,
             NetworkingConstants.kSortOrder: NetworkingConstants.kDescending
         ]
+
+        if let filter = filter {
+            for (key, value) in filter {
+                params[key] = value
+            }
+        }
+
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/AdminOperations.swift:143]: should probably make a helper function that adds key value pairs to the params array so we don't have to do it again everytime

File: sage-ios/SAGE/SAGE/AnnouncementsViewController.swift [hunk 2]
```diff
@@ -121,9 +140,55 @@ class AnnouncementsViewController: UITableViewController {
...
+        let menuController = MenuController(title: "Filter Options")
+        menuController.addMenuItem(MenuItem(title: "All", handler: { (_) -> Void in
+            self.filter = nil
+            self.getAnnouncements(reset: true)
+            self.titleView.setSubtitle("All")
+        }))
+
+        if LoginOperations.getUser()!.isDirector() {
+            menuController.addMenuItem(MenuItem(title: "My School", handler: { (_) -> Void in
+                self.filter = [AnnouncementConstants.kSchoolID: String(LoginOperations.getUser()!.directorID)]
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/AnnouncementsViewController.swift]: isn't director id nil if the user is not a director?
so you'd want to do LoginOperations.getSchool()!.id instead

File: sage-ios/SAGE/SAGE/AnnouncementsViewController.swift [hunk 3]
```diff
@@ -121,9 +140,55 @@ class AnnouncementsViewController: UITableViewController {
...
+                self.titleView.setSubtitle("My School")
+            }))
+        } else if let userSchool = KeychainWrapper.objectForKey(KeychainConstants.kSchool) as? School {
+            menuController.addMenuItem(MenuItem(title: "My School", handler: { (_) -> Void in
+                self.filter = [AnnouncementConstants.kSchoolID: String(userSchool.id)]
+                self.getAnnouncements(reset: true)
+                self.titleView.setSubtitle(userSchool.name!)
+            }))
+        }
+
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/AnnouncementsViewController.swift:165]: for directors, isn't their school also the id of the school in the keychain -- so the else if statement will work for all cases?

File: sage-ios/SAGE/SAGE/CheckinRequestsViewController.swift [hunk 4]
```diff
@@ -7,13 +7,20 @@
 //
 
 import UIKit
+import FontAwesomeKit
 
 class CheckinRequestsViewController: UITableViewController {
+
     var requests: [Checkin]?
+    var filter: [String: AnyObject]?
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/CheckinRequestsViewController.swift:15]: we should probably add this property to sgtableviewcontroller later

File: sage-ios/SAGE/SAGE/CheckinRequestsViewController.swift [hunk 5]
```diff
@@ -42,19 +52,31 @@ class CheckinRequestsViewController: UITableViewController {
...
         super.viewDidLoad()
-        self.title = "Check In Requests"
+        self.navigationItem.titleView = self.titleView
         self.tableView.tableFooterView = UIView()
         
+        let filterIcon = FAKIonIcons.androidFunnelIconWithSize(UIConstants.barbuttonIconSize)
+        let filterImage = filterIcon.imageWithSize(CGSizeMake(UIConstants.barbuttonIconSize, UIConstants.barbuttonIconSize))
+        let filterButton = UIBarButtonItem(image: filterImage, style: .Plain, target: self, action: "showFilterOptions")
+        self.navigationItem.rightBarButtonItem = filterButton
+
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/CheckinRequestsViewController.swift:67]: these can probably be added to sgtableviewcontroller later too

File: sage-ios/SAGE/SAGE/CheckinRequestsViewController.swift [hunk 6]
```diff
@@ -81,6 +112,42 @@ class CheckinRequestsViewController: UITableViewController {
...
+        menuController.addMenuItem(MenuItem(title: "None", handler: { (_) -> Void in
+            self.filter = nil
+            self.loadCheckinRequests(reset: true)
+            self.titleView.setSubtitle("All")
+        }))
+
+        if LoginOperations.getUser()!.isDirector() {
+            menuController.addMenuItem(MenuItem(title: "My School", handler: { (_) -> Void in
+                self.filter = [AnnouncementConstants.kSchoolID: String(LoginOperations.getUser()!.directorID)]
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/CheckinRequestsViewController.swift]: CheckinConstants.kSchoolID

File: sage-ios/SAGE/SAGE/SchoolOperations.swift [hunk 7]
```diff
@@ -11,15 +11,23 @@ import AFNetworking
...
+        manager.GET(StringConstants.kEndpointGetSchools, parameters: nil, success: { (operation, data) -> Void in
+            var schools = [School]()
+            let schoolArray = data["schools"] as! [AnyObject]
+            for schoolDict in schoolArray {
+                let school = School(propertyDictionary: schoolDict as! [String: AnyObject])
+                schools.append(school)
+            }
+            schools.sortInPlace({ (school1, school2) -> Bool in
+                school1.isBefore(school2)
+            })
```
Comment (sameerav) [inline sage-ios/SAGE/SAGE/SchoolOperations.swift:25]: shouldn't these be sorted by the sortAttr stuff on the backend

File: sage-ios/SAGE/SAGE/AnnouncementsViewController.swift (hunk 3 above)
Comment (amillman) [inline sage-ios/SAGE/SAGE/AnnouncementsViewController.swift:165]: there can be some corner cases for directors if they change their school, but we wanna prioritize the school they direct

File: sage-ios/SAGE/SAGE/CheckinRequestsViewController.swift (hunk 4 above)
Comment (amillman) [inline sage-ios/SAGE/SAGE/CheckinRequestsViewController.swift:15]: that is a good idea. definitely should do dat

File: sage-ios/SAGE/SAGE/SchoolOperations.swift (hunk 7 above)
Comment (amillman) [inline sage-ios/SAGE/SAGE/SchoolOperations.swift:25]: its weird cuz its all done by 1 query for checkins. for verified, we wanna sort by most recent start time. but for unverified we sort by most recent creation time. then again for verified i think the order will stay the same if we order by creation time

File: sage-ios/SAGE/SAGE/AdminOperations.swift (hunk 1 above)
Comment (amillman) [inline sage-ios/SAGE/SAGE/AdminOperations.swift:143]: done

**Pull request merged**